internal/command/deploy/deploy.go

```diff
@@ -155,13 +155,26 @@ func DeployWithConfig(ctx context.Context, appConfig *appconfig.Config, args Dep
 	case err != nil:
 		return err
 	case isV2App:
-		if err := appConfig.EnsureV2Config(); err != nil {
+		if err = appConfig.EnsureV2Config(); err != nil {
 			return fmt.Errorf("Can't deploy an invalid v2 app config: %s", err)
 		}
-		return deployToMachines(ctx, appConfig, appCompact, img)
+		err = deployToMachines(ctx, appConfig, appCompact, img)
+		if err != nil {
+			return err
+		}
 	default:
-		return deployToNomad(ctx, appConfig, appCompact, img)
+		err = deployToNomad(ctx, appConfig, appCompact, img)
+		if err != nil {
+			return err
+		}
 	}
+
+	url, err := appConfig.URL()
+	if err == nil {
```

Comment:
the only extra check I would do her is `url != ""` in case the app have no http services